src/homeworkTenth/MyHashMap.java

```diff
@@ -0,0 +1,81 @@
+package homeworkTenth;
+
+public class MyHashMap<K, V> {
+
+    private Entry<K, V>[] buckets;
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private int size = 0;
+
+    public MyHashMap() {
+        this(INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int capacity) {
+        this.buckets = new Entry[capacity];
+    }
+
+    public void put(K key, V value) {
+        Entry<K, V> entry = new Entry<>(key, value, null);
+        int bucket = getHash(key) % getBucketSize();
+
+        Entry<K, V> exiting = buckets[bucket];
+        if (exiting == null) {
+            buckets[bucket] = entry;
+            size++;
+        } else {
+            while (exiting.next != null) {
```

Comment:
may be change with second condition?

Reply:
no. 
here we compare the keys if the key already exists